SMLHelper/Assets/Spawnable.cs

```diff
@@ -70,22 +86,33 @@ public abstract class Spawnable : ModPrefab
         protected Spawnable(string classId, string friendlyName, string description)
             : base(classId, $"{classId}Prefab")
         {
-            if (string.IsNullOrEmpty(classId))
+            if(string.IsNullOrEmpty(classId))
             {
                 Logger.Log($"ClassID for Spawnables must be a non-empty value.", LogLevel.Error);
                 throw new ArgumentException($"Error patching Spawnable");
             }
 
-            this.FriendlyName = friendlyName;
-            this.Description = description;
+            FriendlyName = friendlyName;
+            Description = description;
 
             CorePatchEvents += () =>
             {
-                this.PrefabHandler.RegisterPrefab(this);
-                this.SpriteHandler.RegisterSprite(this.TechType, GetItemSprite());
+                PrefabHandler.RegisterPrefab(this);
+                SpriteHandler.RegisterSprite(TechType, GetItemSprite());
+
+                if(!SizeInInventory.Equals(defaultSize))
+                {
+                    CraftDataHandler.SetItemSize(TechType, SizeInInventory);
+                }
+            };
 
-                if (!this.SizeInInventory.Equals(defaultSize))
-                    this.CraftDataHandler.SetItemSize(this.TechType, this.SizeInInventory);
+            OnFinishedPatching += () =>
+            {
+                if(EntityInfo != null && BiomesToSpawnIn != null)
```

Comment:
Is there any reason why this needs to be done in `OnFinishedPatching` and can't be done from `CorePatchEvents`

Reply:
Actually no. I don't think so.

Reply:
Then it would be best to put it inside the `CorePatchEvents` since that is `internal` and managed by SMLHelper while `OnFinishedPatching` is `public` and handled by the individual mods.

Reply:
Will do.